terraform/app/modules/paas/main.tf

```diff
@@ -25,6 +25,52 @@ resource "cloudfoundry_user_provided_service" "papertrail" {
   syslog_drain_url = var.papertrail_url
 }
 
+resource "aws_s3_bucket" "documents_s3_bucket" {
+  bucket = local.documents_s3_bucket_name
+}
+
+resource "aws_s3_bucket_public_access_block" "documents_s3_bucket_block" {
+  bucket = aws_s3_bucket.documents_s3_bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_iam_policy" "documents_s3_bucket_policy" {
+  name   = "${local.documents_s3_bucket_name}-policy"
+  path   = "/attachment_buckets_policies/"
+  policy = data.aws_iam_policy_document.documents_s3_bucket_policy_document.json
+}
+
+data "aws_iam_policy_document" "documents_s3_bucket_policy_document" {
+  statement {
+    actions   = ["s3:ListAllMyBuckets"]
```

Comment:
Shall we add `s3:ListBucket` to be able to list objects in the bucket?
See [ListBucket IAM permission](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-actions-as-permissions) and [ListObjectsV2 command](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html)

Reply:
I _think_ we won't need to list objects in the bucket, so happy to not include it (`GetObject` is enough). I think `ListAllMyBuckets` is required because Terraform fails otherwise.